### PR TITLE
ConvertThingToResource - added the ability to use <thingCategoryDef> in addition to <thingDef>.

### PR DESCRIPTION
**Pull request by DestinyPlayer**
ConvertThingToResource - added the ability to use thingCategoryDef in addition to thingDef.

This allows you to convert entire categories of things instead of just a specific thing.

I needed the ability to filter for an entire category in my own mod, and unfortunately you can't easily add your own comps to PipeSystem from your own mod. As a result, this was really the best I could do, request a pull with what I need and hope that it gets accepted.

### CODE REVIEW

File: Source/PipeSystem/PipeSystem/Comps/CompProperties/CompProperties_ConvertThingToResource.cs [hunk 1]
```diff
@@ -23,8 +24,10 @@ public override IEnumerable<string> ConfigErrors(ThingDef parentDef)
                 yield return "Can't use CompProperties_ConvertThingToResource with a thing that don't have Building_Storage as thingClass.";
             if (parentDef.comps.FindAll(c => c is CompProperties_ConvertThingToResource).Count > 1)
                 yield return "Can't use multiple CompProperties_ConvertThingToResource on the same thing.";
-            if (thing == null)
-                yield return "Can't use CompProperties_ConvertThingToResource with a null thing.";
+            if (thing == null & category == null)
+                yield return "Can't use CompProperties_ConvertThingToResource with a null thing or category.";
+            if (thing != null & category != null)
+                yield return "Can't use both thing and category on the same thing.";
```
Comment (KylianB) [inline Source/PipeSystem/PipeSystem/Comps/CompProperties/CompProperties_ConvertThingToResource.cs:30]: Would be easier to understand: "Can't use both thing and category on the same CompProperties_ConvertThingToResource."

File: Source/PipeSystem/PipeSystem/Comps/CompConvertToResource.cs [hunk 2]
```diff
@@ -34,10 +34,29 @@ public Thing HeldThing
...
+                if (Props.category != null)
+                {
+                    List<ThingDef> categoryList = Props.category.childThingDefs;
+                    for (int j = 0; j < categoryList.Count; j++)
+                    {
+                        for (int i = 0; i < thingList.Count; i++)
+                        {
+                            if (thingList[i].def == categoryList[j])
+                            {
+                                Log.Message("Eating " + thingList[i].ToString() + " from category " + Props.category.ToString());
```
Comment (KylianB) [inline Source/PipeSystem/PipeSystem/Comps/CompConvertToResource.cs:47]: Please comment out the logging (useful for dev, but can annoy players)

**Review by KylianB — CHANGES_REQUESTED** [review]
Thank you for the PR. 
It looks excellent, I'm only requesting two small change before merging.

File: Source/PipeSystem/PipeSystem/Comps/CompProperties/CompProperties_ConvertThingToResource.cs (hunk 1 above)
Comment (DestinyPlayer) [reply Source/PipeSystem/PipeSystem/Comps/CompProperties/CompProperties_ConvertThingToResource.cs]: Agreed. Phrasing changed.

File: Source/PipeSystem/PipeSystem/Comps/CompConvertToResource.cs (hunk 2 above)
Comment (DestinyPlayer) [reply Source/PipeSystem/PipeSystem/Comps/CompConvertToResource.cs]: That was on me, forgot to change that after depths of testing.

Commented out.

**Review by KylianB — APPROVED** [review]
Thank you, everything look good now

**Pull request merged**